packages/headless/src/features/folding/folding-actions-loader.ts

```diff
@@ -33,10 +34,10 @@ export interface FoldingActionCreators {
    * @returns A dispatchable action.
    */
   loadCollection(
-    collectionId: string
+    result: ResultWithFolding
```

Comment:
This is a breaking change on the public API: a publicly exported function from the package which used to accept a `string` as a parameter now accept a `ResultWithFolding`. 

User code importing and trying to use this function, on upgrade, would either no longer compile (if they are using TypeScript), or would crash at runtime if using JavaScript.

We unfortunately cannot do this.

We can either:
* Add a new function with a different name (and potentially mark the old one as `@deprecated`, but still functional)
* expand the accepted parameter type to be `string | ResultWithFolding` and have logic to detect which it is
* Or see if there's another solution that does not involve modifying the public API

Reply:
hmm I was thinking this would then be a breaking change that would bump the major version...? but I guess that's not how it works!
@btaillon since you understand a bit more what's going on with these results, I'll let you give your opinion on whether this is the way to go before I make any changes!

Reply:
> hmm I was thinking this would then be a breaking change that would bump the major version

Yes that's what we do for breaking changes, but we don't bump major versions often, because it's annoying to the lib users to upgrade and it's hard for us to maintain. We are planning a major bump this summer I think? There will be a laundry list of improvements to create and we can do them all at once when going to v2!